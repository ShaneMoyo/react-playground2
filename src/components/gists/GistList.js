import React from 'react'; 
import {
    useParams,
    Link
  } from "react-router-dom";

export default function GistList({ gists }) {
    const gistItem = gist => (
        <li>
            <Link to={`/gist/${gist.id}`}>
                <span>
                    <p>{gist.description}</p>
                </span>
            </Link>
        </li>
    );
    
    const gistItems = gists ? gists.map(gist => gistItem(gist)) : null;

    return (
        <ul>
            {gistItems}
        </ul>
    )
}