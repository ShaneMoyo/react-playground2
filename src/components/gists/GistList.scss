* {
    box-sizing: border-box;
  }
  
  html,
  body {
    margin: 0;
    padding: 0;
  }
  
  body {
    padding: 20px;
  }h1 {
    text-align: center;
  }
  
  
  
  table {
    width: 100%;
  }
  
  td,
  th {
    text-align: center;
    border: solid 1px #ccc;
    font-size: 22px;
    padding: 6px;
    word-wrap: break-word;
  }
  
  td {
    color: red;
    font-weight: bold;
  }
  
  .check {
    color: limegreen;
    font-size: 30px;
  }
  
  th:first-child,
  td:first-child {
    font-weight: bold;
    color: black;
  }
  
  @media screen and (max-width: 768px) {
    .check {
      color: limegreen;
      font-size: 20px;
    }
    tr {
      display: flex;
      flex-flow: row wrap;
      justify-content: space-around;
    }
    td,
    th {
      display: block;
      width: 33.333333333333%;
      font-size: 20px;
    }
    th:first-child,
    td:first-child {
      background: #efefef;
      width: 100%;
    }
    th:first-child {
      display: none;
    }
  }
  
  .p {
    text-align: center;
    font-size: 13px;
    padding-top: 130px;
  }