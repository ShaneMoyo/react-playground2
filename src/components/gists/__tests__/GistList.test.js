import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import GistList from '../GistList'; 



describe('components', () => {
   
  it('should render AddImage component', () => {
    const wrapper = shallow(<GistList/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});