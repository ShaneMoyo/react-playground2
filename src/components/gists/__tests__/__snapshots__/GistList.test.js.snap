// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components should render AddImage component 1`] = `<ul />`;
